### DISCUSSION TITLE
Velocity not Applied Properly in Moving Wall Channel Flow

### DISCUSSION THREAD

**Discussion by wwdhd** [xcompact3d/Incompact3d#276] [Q&A] [answered]
Dear all,

I am trying to apply the translating wall in the flow. I applied z velocity on the top wall, which I assigned at byzn value (I added the variable manually) in Case-Channel.f90 with a value of 0.8. However, plotting the result, the z velocity near the bottom of the wall is zero, with maximum near the flow is about -0.017. Is there any steps missing from my methods? Is it because of the dpdxy correction in pres_correc that keeps the velocity low? Thank you in advance.

Wid

**Comment by slaizet** [comment]
Hi,
I am not quite sure to understand what you are doing. You impose a z velocity at the top of a channel flow and you expect the z velocity at the bottom of the channel to -not- be zero?
Best,
Sylvain

**Reply by wwdhd** [reply]
Dear Sylvain,

Sorry for the confusing sentences. I mean not to be zero at the top of the channel. Let me rephrase the question:

So I am trying to impose a z velocity of 0.8 on the top wall of a channel flow. I changed the velocity by adding the byzn variable in Case-Channel.f90. However, at the top channel, the resulting w velocity at the wall is zero, with maximum near the flow is about -0.017 when plotted in Paraview. Is there any way to prevent that? 

edit: I tried forcing it into the uz(i, xsize(3), k) and uz(i, xsize(2), k) equation in pres_correc (navier.f90)  by substituting the byzn value to the actual value (0.8). I tried to print the uz(1,xsize(2),1) value in the .log file on every iteration and interestingly enough the value starts with 0.10560397934865830 and decreasing as the time goes to zero. I don't know why.

**Reply by slaizet** [reply]
I am going to investigate but can I just check that you are using a fully explicit time scheme?
Cheers,
Sylvain

**Reply by wwdhd** [reply]
Yes, I used explicit time scheme (iimplicit = 0). I also used implicit time scheme (Crank-Nicholson, iimplicit = 2) which was the default on the input.i3d.

Regards,
Wid

**Comment by slaizet** [comment, marked as answer]
Using an explicit time advancement scheme (RK3), I imposed uz(i,xsize(2),k)=0.1 in the pre_correc subroutine (for nclyn=2) and the code is behaving as expected (the velocity uz is equal to 0.1 at the top wall for the all simulation).
Best,
Sylvain
